url354ce1a2d1_dagaren_gladiator_chess_tgz-pJ8-es_dagaren_gladiator_main_MainJ8/src/es/dagaren/gladiator/representation/AbstractPosition.java

```diff
@@ -0,0 +1,856 @@
+      if(ranksStrings.length != 8)
+         throw new RuntimeException("Posición fen incorrecta: "+ fenPosition);
+      
+      for(int i=0;i<8;i++)
+      {
+         String rankString = ranksStrings[i];
+         
+         int file = 0;
+         
+         for(int j=0;j<rankString.length();j++)
+         {
+            char character = rankString.charAt(j);
+            
+            if(file > 8)
+               throw new RuntimeException("Posición fen incorrecta: " + fenPosition);
+            
+            switch(character)
+            {
+                case 'p':
+                  putGenericPiece(GenericPiece.PAWN,Colour.BLACK,Square.fromRankAndFile(7-i, file++));
+                    break;
+                case 'r':
+                    putGenericPiece(GenericPiece.ROOK,Colour.BLACK,Square.fromRankAndFile(7-i, file++));
+                    break;
+                case 'n':
+                    putGenericPiece(GenericPiece.KNIGHT,Colour.BLACK,Square.fromRankAndFile(7-i, file++));
+                    break;
+                case 'b':
+                    putGenericPiece(GenericPiece.BISHOP,Colour.BLACK,Square.fromRankAndFile(7-i, file++));
+                    break;
+                case 'q':
+                    putGenericPiece(GenericPiece.QUEEN,Colour.BLACK,Square.fromRankAndFile(7-i, file++));
+                    break;
+                case 'k':
+                    putGenericPiece(GenericPiece.KING,Colour.BLACK,Square.fromRankAndFile(7-i, file++));
+                    break;
+                case 'P':
+                    putGenericPiece(GenericPiece.PAWN,Colour.WHITE,Square.fromRankAndFile(7-i, file++));
+                    break;
+                case 'R':
+                    putGenericPiece(GenericPiece.ROOK,Colour.WHITE,Square.fromRankAndFile(7-i, file++));
+                    break;
+                case 'N':
+                    putGenericPiece(GenericPiece.KNIGHT,Colour.WHITE,Square.fromRankAndFile(7-i, file++));
+                    break;
+                case 'B':
+                    putGenericPiece(GenericPiece.BISHOP,Colour.WHITE,Square.fromRankAndFile(7-i, file++));
+                    break;
+                case 'Q':
+                    putGenericPiece(GenericPiece.QUEEN,Colour.WHITE,Square.fromRankAndFile(7-i, file++));
+                    break;
+                case 'K':
+                    putGenericPiece(GenericPiece.KING,Colour.WHITE,Square.fromRankAndFile(7-i, file++));
+                    break;
+                default:
+                    try
+                    {
+                        int number = Integer.parseInt(rankString.substring(j,j+1));
+
+                        if(number < 1 || number > 8)
+                           throw new RuntimeException("Posición fen incorrecta: " + fenPosition);
+                        else
+                        {
+                            file += number;
+                        }
+                    }
+                    catch(Exception ex)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 419, 430.